internal/pkg/cache/cache_test.go

```diff
@@ -40,13 +49,116 @@ func testPackCount(t *testing.T, opts cacheOperationProvider) int {
 	return packCount
 }
 
+func TestMain(m *testing.M) {
+	cleanup := makeTestRegRepo()
+	exitCode := m.Run()
+	cleanup()
+	os.Exit(exitCode)
+}
+
+func makeTestRegRepo() func() {
+	dirName, err := os.MkdirTemp("", "cache-test-*")
+	if err != nil {
+		panic(err)
+	}
+	cleanup := func() { os.RemoveAll(dirName) }
+	maybeFatal := func(err error) {
+		if err != nil {
+			var out string
+
+			switch err := err.(type) {
+			case *exec.ExitError:
+				out = fmt.Sprintf("Error: %v \nStdErr:\n%s", err, string(err.Stderr))
+			default:
+				out = fmt.Sprintf("Error: (%T) %v ", err, err)
+			}
+
+			cleanup()
+			panic(out)
+		}
+	}
+	// fmt.Printf("Temp Dir: %s\n", dirName)
+
+	*testRegistryURL = path.Join(dirName, "test_registry.git")
+	maybeFatal(filesystem.CopyDir("../../../fixtures/test_registry", *testRegistryURL, NoopLogger{}))
+	var exitErr *exec.ExitError
+
+	formatPanic := func(res GitCommandResult) {
+		var out strings.Builder
+		out.WriteString(fmt.Sprintf("git err: %v running %v\n", res.err.Error(), res.cmd))
+		out.WriteString(fmt.Sprintf("stdout:\n%s\nstderr:\n%s\n", res.stdout, res.stderr))
+		// If these setup git commands fail, there's no use in continuing
+		// because almost all of the cache tests will fail. Could these
+		// be refactored into a sync.Once and a check for cache tests?
+		cleanup()
+		panic(out)
+	}
+
+	handleInitError := func(res GitCommandResult) {
+		if res.err != nil && errors.As(res.err, &exitErr) && !strings.Contains(
+			res.stdout,
+			"Initialized empty Git repository",
+		) {
+			formatPanic(res)
+		}
+	}
+
+	handleGitError := func(res GitCommandResult) {
+		if res.err != nil {
+			formatPanic(res)
+		}
+	}
+
+	git(handleInitError, "init")
+	git(handleGitError, "config", "user.email", "test@example.com")
+	git(handleGitError, "config", "user.name", "Github Action Test User")
+	git(handleGitError, "add", ".")
+	git(handleGitError, "commit", "-m", "Initial Commit")
+	res := git(handleGitError, "log", "-1", `--pretty=%H`)
+	*testRegistryRef1 = strings.TrimSpace(res.stdout)
+
+	git(handleGitError, "commit", "--allow-empty", "-m", "Second Commit")
+	res = git(handleGitError, "log", "-1", `--pretty=%H`)
+	*testRegistryRef2 = strings.TrimSpace(res.stdout)
```

Comment:
Personal preference, but passing the error handler like this feels a bit odd to me. A more explicit error handling would look like

```go
res := git("init")
if res != nil {
    handleInitError(res)
}
res := git(, "config", "user.email", "test@example.com")
if res != nil {
    handleGitError(res)
}
// ...
```

But that's quite verbose. Another option would be to make `git` a stateful `struct` that you can just call and it would store errors as they happen:

```go
type git struct {
    res *GitCommandResult
}

func (g git) run(args ...string) GitCommandResult {
    if g.res != nil {
        // we've failed already, so noop
        return nil
    }
    git := exec.Command("git", args...)
    // ...
    err := git.Run()
    if err != nil {
        g.res = GitCommandResult { ... }
    }
}

func makeTestRegRepo() func() {
    // ...
    g := git{}
    g.run("init")
    if g.res != nil {
        // handle init error
    }

    g.run("config", "user.email", "test@example.com")
    g.run("config", "user.name", "Github Action Test User")
    g.run("add", ".")
    // ...
    if g.res != nil {
        formatPanic(res)
    }
    // ..
```